pkg/trace/stats/concentrator_test.go

```diff
@@ -410,13 +489,161 @@ func TestConcentratorSublayersStatsCounts(t *testing.T) {
 		"query|hits|env:none,resource:resource4,service:A3":                                               1,
 		"query|hits|env:none,resource:resource6,service:A3":                                               1,
 	}
+	countValsEq(t, expectedCountValByKey, receivedCounts)
+}
+
+// setupConcentrator initializes a concentrator from a given trace input.
+// calcSublayers is an option to enable sublayer metric calculation.
+func setupConcentrator(trace pb.Trace, calcSublayers bool) *Concentrator {
+	statsChan := make(chan []Bucket)
+	traceutil.ComputeTopLevel(trace)
+	wt := NewWeightedTrace(trace, traceutil.GetRoot(trace))
+	testTrace := &Input{
+		Env:   "none",
+		Trace: wt,
+	}
+	if calcSublayers {
+		subtraces := ExtractTopLevelSubtraces(trace, traceutil.GetRoot(trace))
+		sublayers := make(map[*pb.Span][]SublayerValue)
+		for _, subtrace := range subtraces {
+			subtraceSublayers := ComputeSublayers(subtrace.Trace)
+			sublayers[subtrace.Root] = subtraceSublayers
+		}
+		testTrace.Sublayers = sublayers
+	}
+	c := NewConcentrator([]string{}, testBucketInterval, statsChan)
+	c.addNow(testTrace, time.Now().UnixNano())
+	return c
+}
+
+var addNowTests = []struct {
```

Comment:
For the new `TestConcentratorAddNow` and `TestConcentratorMeasuredSpanSublayers` tests I define the tests up front and use the table-driven approach.

Let me know if they could be more idiomatic.